components/nimbus/src/enrollment.rs

```diff
@@ -1049,14 +1073,56 @@ pub fn map_features_by_feature_id(
     let (experiments, exp_enrollments) =
         filter_experiments_and_enrollments(experiments, enrollments, |exp| !exp.is_rollout());
 
-    let features_under_rollout = map_features(&ro_enrollments, &map_experiments(&rollouts));
-    let features_under_experiment = map_features(&exp_enrollments, &map_experiments(&experiments));
+    let features_under_rollout = map_features(
+        &ro_enrollments,
+        &map_experiments(&rollouts),
+        coenrolling_ids,
+    );
+    let features_under_experiment = map_features(
+        &exp_enrollments,
+        &map_experiments(&experiments),
+        coenrolling_ids,
+    );
 
     features_under_experiment
         .defaults(&features_under_rollout)
         .unwrap()
 }
 
+fn populate_feature_maps(
+    enrolled_feature: EnrolledFeatureConfig,
+    coenrolling_feature_ids: &HashSet<&str>,
+    enrolled_features: &mut HashMap<String, EnrolledFeatureConfig>,
+    coenrolling_features: &mut HashMap<String, EnrolledFeatureConfig>,
+) {
+    let feature_id = &enrolled_feature.feature_id;
+    if !coenrolling_feature_ids.contains(feature_id.as_str()) {
+        // If we're not allowing co-enrollment for this feature, then add it to enrolled_features.
+        // We'll use this map to prevent collisions.
+        enrolled_features.insert(feature_id.clone(), enrolled_feature);
+    } else if let Some(existing) = coenrolling_features.get(feature_id) {
+        // Otherwise, we'll add to the coenrolling_features map.
+        // In this branch, we've enrolled in one experiment already before this one.
+        // We take care to merge this one with the existing one.
+        let merged = enrolled_feature
+            .defaults(existing)
+            .expect("A feature config hasn't been able to merge; this is a bug in Nimbus");
```

Comment:
Do we have it set up such that the order in which enrolled feature configs will enter this method will be the same across all clients/enrollment cases? Thinking about the case where two experiments set different values for the same feature variable — wondering if each client could end up seeing different things as a result of that

Reply:
I made a change to `evolve_enrollment_recipes` to make it more deterministic, though I'm not yet ready to declare the SDK completely deterministic. However, in practice, I think this is for the app features to define FML schema that will merge well.